doc/feature/ReqRep.md

```diff
@@ -0,0 +1,112 @@
+Request-Reply {#reqrep}
+============
+
+Certain operations are by necessity synchronous, RPC-like exchanges. This
+specification describes this client-server API based on zmq req-rep (dealer/router) sockets.
+
+## Requirements
+
+* Asynchronous remote procedure call from a Client to a Server
+
+## API
+
+    using ReplyFunc = std::function<void(const void*, size_t)>;
+
+    class Client : public Receiver
+    {
+        // ctors same as Subscriber
+        void get(const Serializable& request, ReplyFunc& func);
```

Comment:
That's not entirely satisfactory. In a request that has an empty reply (item 3 in the list from Raphael) how can I distinguish a success from a failure?